website/content/docs/concepts/oidc-provider.mdx

```diff
@@ -0,0 +1,144 @@
+
+The following defines the claims key and value mapping for the openid scope:
+
+* `iss`- configured issuer of the provider
+* `sub`- unique entity ID of the Vault user
+* `aud`- iD of the client
+* `iat`- time of token issue
+* `exp`- time of token issue + ID token TTL
+
+### Client registration
+
+A client resource allows the relying party by providing metadata about itself to Vault to [dynamically register](https://openid.net/specs/openid-connect-registration-1_0.html).
+
+The client must have the following configuration parameters:
+
+* **Redirect URIs**: limits the valid redirect URIs in an authentication request
+* **Assignments**: determines who can authenticate with the client
+* **Key**: signing of the ID tokens
+*	**ID token TTL**: specifies the time-to-live for ID tokens
+* **Access token TTL**: establishes the time-to-live for access tokens
+
+A `client_id` and `client_secret` are generated and returned after a successful client registration. Their values are strings using the base62 character set. The `client_id` will have 32 characters, and the `client_secret` will have a prefix of `hvo_secret`. The `client_id`  uniquely identifies the client. The `client_secret` will be used to authenticate to the token endpoint as described in [client authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication).
+
+The `key` parameter is required. The user must create a `key` as a required parameter of the client configuration.
+
+~> **Note**: At least one of the redirect URIs of a client must exactly match the `redirect_uri` parameter used in an authentication request initiated by the client.
+
+### Assignments
+
+Assignment resources are referenced by clients via the `assignments` parameter. This parameter limits the set of Vault users allowed to authenticate. The assignments of an associated client are validated during the authentication request, ensuring that the Vault identity associated with the request is a member of the assignment's entities or groups.
+
+### Keys
+
+Key resources are referenced by clients via the key parameter. This parameter specifies the key that will be used to sign ID tokens for the client. The key resource established is used to support identity tokens. See existing [documentation](/api-docs/secret/identity/tokens#create-a-named-key) for details on keyring management, supported signing algorithms, rotation periods, and verification TTLs. Currently, a key referenced by a client cannot be changed.
+
+## OIDC flow
+
+The following sections provide implementation details for the OIDC compliant APIs provided by Vault OIDC providers.
```

Comment:
I don't know if it makes sense to go here or at the top of this document. But we should clarify that the Vault OIDC Provider feature currently only supports the [authorization code flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).

Below are some more details that should be mentioned in regards to the OIDC flow:

Vault OIDC providers will enable registered clients to authenticate and obtain identity information (or "claims") for their end-users. They will do so by providing the APIs and behavior required to satisfy the OIDC specification for the [authorization code flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth). Initially, all clients will be treated as first-party. This means that end-users will not be required to provide consent to the provider as detailed in [3.1.2.4](https://openid.net/specs/openid-connect-core-1_0.html#Consent) of the OIDC specification. The provider will release information to clients as long as the end-user has ACL access to the provider and their identity has been authorized via an assignment.

Reply:
Make sense. I added a note at the top of the document to specify that the feature is currently only supports the authorization code flow.

I also expanded the OIDC flow section to include your suggested text above. I've changed the tense a bit from passive to active in some places where I thought it might help. Sometimes, it's unavoidable and using passive voice might be necessary.

Reply:
thanks!